go-controller/pkg/ovn/controller/services/services_controller_test.go

```diff
@@ -510,6 +510,120 @@ func TestSyncServices(t *testing.T) {
 	}
 }
 
+func Test_ETPCluster_NodePort_Service_WithMultipleIPAddresses(t *testing.T) {
```

Comment:
thank you for the tests++ nit: suggestion to swap some into IPV6 addresses to cover those bits as well but I won't hold the PR for that.

Reply:
I added a commit to convert that test to dualstack:
https://github.com/ovn-org/ovn-kubernetes/pull/3557/commits/8a973fa83877a2609bcdcd6832c5fde0c805d27c

Test code becomes a little bit more complicated. WDYT?

Reply:
I am good with this thanks @zeeke !